src/modules/network/components/networkSwitcherDropdown/NetworkSwitcherDropdown.js

```diff
@@ -52,15 +53,21 @@ function NetworkSwitcherDropdown({ noLabel, onNetworkSwitchSuccess }) {
       queryClient.current.create({
         http: network.serviceUrl,
       });
-      networkStatus.refetch().then((res) => {
+      setSelectedNetwork(network);
+    },
+    [networkStatus]
+  );
+
+  useEffect(() => {
+    if (selectedNetwork.serviceUrl) {
+      networkStatus?.refetch?.()?.then((res) => {
```

Comment:
There seems to be a maximum call stack exceeded error when I click on the network `DialogNetworkApplicationSelector`. It seems because this is now in a useEffect, there are states scenarios that un-wantedly triggers the setValue from setting and since the deepMerge logic is reclusive it could have been called a lot of time resulting lots of calls to deepMerge (this is an assumption though).

<img width="1315" alt="image" src="https://github.com/LiskHQ/lisk-desktop/assets/32064769/134c56fc-eaf5-4c8e-b684-475a1e8903dc">